### ISSUE TITLE
Hang on python docstrings: Company: backend company-capf error "Wrong type argument: arrayp, nil" with args (candidates """

### ISSUE THREAD

**Issue by butala** [company-mode/company-mode#571]
Hello,

I use a fairly vanilla install of prelude which enables company by default. I am hitting the following problem. I tend to write long docstrings in django style (see `python-fill-docstring-style`), for example:

```
def foo(x):
   """
   This is the django style for docstrings: triple quotes \n, docstring content broken 
   at column 80 across multiple lines, and then \n triple quotes \n.
   """
   return x
```

When I type in the docstring like the above, emacs will eventually hang and I have to `C-g` and `*Messages*` reports

```
Company: An error occurred in auto-begin
Company: backend company-capf error "Wrong type argument: arrayp, nil" with args (candidates """
    This is a dfi)
Quit
```

If I turn off company mode, no hangs. If I write in the pep-257 style, e.g.,

```
def foo(x):
   """I am a PEP-257 docstring --- triple quote, no line break, and then triple quote."""
   return x
```

I experience no hangs. This behavior changed somewhat recently (my django style docstrings previously caused no problems).

Is there some way to debug this further? I very much like company mode, but must turn it off while developing python code.

--- Mark

**Comment by dgutov** [comment]
Hi!

Do you see the same error if you just press `C-M-i`? If so, please `M-x report-emacs-bug`.

**Comment by butala** [comment]
When I am within a docstring, `C-M-i` does nothing. When outside, it does what I expect (brings up the completion dialog box). Would you like to see `M-x report-emacs-bug` anyways?

Thank you for the help!

**Comment by dgutov** [comment]
Could you try it in the exact same context that you get the freeze? Like, you see it free, hit `C-g` to unfreeze it, and try `C-M-i`.

**Comment by butala** [comment]
The behavior is the following. Start typing in docstring, pause, emacs freezes. I hit `C-g` to unfreeze and then immediately `C-M-i`. The interesting thing is that it freezes again immediately after I hit `C-M-i` --- normally I would have to pause a little after typing to get the freeze (I assume for the completion timeout to trigger).

**Comment by dgutov** [comment]
> (I assume for the completion timeout to trigger)

Right.

Now, with the same buffer contents, does `C-M-i` make Emacs freeze, even if `company-mode` is disabled? Say, you saved the file in "unfinished" state, and then close Emacs, reopen it, go to the same position and press `C-M-i`.

**Comment by butala** [comment]
I edited my configuration to completely turn off company mode, quit emacs, restarted, and opened the file I was editing. I can confirm that company mode is not enabled (indicator absent on the mode line). Now, at the point where `C-M-i` previously produced a hang, it no longer does with company mode disabled.

**Comment by dgutov** [comment]
Hmm.

Please try the following experiment, then: delete some of the last characters. Then type them out one by one and press `C-M-i` after each one. Does it freeze then?

**Comment by butala** [comment]
With company mode off, no freeze, With company mode on, and, for as fast as I can quickly alternate between typing a character and then hitting `C-M-i`, no freeze.

**Comment by dgutov** [comment]
OK, thanks.

Does it reliably freeze at the same position, after you've typed a specific part of the buffer text? What's your Emacs version?

Please describe a more thorough reproduction scenario, if you can. Or maybe try the Emacs 25.1 RC first; I have it installed, and couldn't reproduce your problem just now.

**Comment by brianhlin** [comment]
I've got the same issue with the following version:

{code}
(version)
"GNU Emacs 24.5.1 (x86_64-redhat-linux-gnu, GTK+ Version 3.18.9) of 2016-04-11 on buildvm-25.phx2.fedoraproject.org"
{code}

The only difference in behavior is that `C-M-i` hangs with company mode on or off. Thanks to OP for pointing out the differences in behavior between django-style and pep257-style docstrings.

**Comment by dgutov** [comment]
> The only difference in behavior is that C-M-i hangs with company mode on or off.

Please `M-x report-emacs-bug` then.

**Comment by sawan** [comment]
Is this issue happening on a Mac computer?

I am having very similar [problems](https://github.com/jorgenschaefer/elpy/issues/1101) but not on Emacs 25.1 on Windows.

**Comment by dgutov** [comment]
@sawan It's not similar. Please avoid spamming the unrelated issues.

**Comment by termlen0** [comment]
Ran into this issue today and arrived at this page. I am running 24.5.1 on ubuntu. as @brianhlin noted, This happens only when I have an inferior python process running.

**Comment by dgutov** [comment]
> This happens only when I have an inferior python process running.

Indeed, that's the necessary condition for the built-in Python completion to work.

Please report Emacs bugs to Emacs. Before that, though, please verify that it still occurs in the latest version. Which is 25.2.
